### PR TITLE
GEODE-4225: Extend ClusterStartupRule to specify member / client version.

### PR DESCRIPTION
**Pull request by PurelyApplied**
* Correct typo: "Embeded" -> "Embedded"
* Reorder class to resemble call heirarchy
* Remove startClient signature inconsistent with other method calls

-----

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/test/dunit/rules/ClusterStartupRule.java [hunk 1]
```diff
@@ -284,39 +302,22 @@ public MemberVM startServerAsEmbededLocator(int index, Properties properties) th
...
-    if (member == null)
-      return;
-
-    member.stopVM(cleanWorkingDir);
+  public MemberVM startServerAsEmbeddedLocator(int index, Properties properties)
+      throws IOException {
+    return startServerAsEmbeddedLocator(index, properties, VersionManager.CURRENT_VERSION);
   }
 
-  public ClientVM startClientVM(int index, String username, String password,
```
Comment (PurelyApplied) [inline geode-core/src/test/java/org/apache/geode/test/dunit/rules/ClusterStartupRule.java]: Whoops.  I had refactored this out during my initial exploration, but had meant to bring it back before opening the PR.

Comment (jinmeiliao) [reply geode-core/src/test/java/org/apache/geode/test/dunit/rules/ClusterStartupRule.java]: yeah, this method would be helpful to stay here.

**Comment by PurelyApplied** [comment]
Some unexpected precheckin failures in translating the VM version shorthand `"000"` depending on environment.  It appears that it can be both `1.x.0-SNAPSHOT` or `0.0.0`.  Will need to adjust the test.

**Comment by PurelyApplied** [comment]
Removed current-version testing, rerunning precheckin.

**Comment by PurelyApplied** [comment]
Precheckin green with current-version testing removed.

**Pull request merged**